pkg/state/state.go

```diff
@@ -322,9 +322,8 @@ CREATE EVENT TRIGGER pg_roll_handle_ddl ON ddl_command_end
 	EXECUTE FUNCTION %[1]s.raw_migration();
 
 DROP EVENT TRIGGER IF EXISTS pg_roll_handle_drop;
-CREATE EVENT TRIGGER pg_roll_handle_drop ON sql_drop
+CREATE EVENT TRIGGER pg_roll_handle_drop ON sql_drop WHEN TAG IN ('DROP TABLE', 'DROP VIEW')
```

Comment:
any chance we are missing other objects that could only be detected here? ie constraints, indices, triggers

Reply:
I've added several more test cases and changed how the trigger function definition so that `raw_migration` runs on all `sql_drop`s but doesn't get the schema for `ALTER TABLE DROP ...` as those will be handled when the function is called for `ddl_command_end`.

I think this way round reduces the chance of missing any operations.